pyparsing/diagram/__init__.py

```diff
@@ -271,6 +278,12 @@ def _worth_extracting(element: pyparsing.ParserElement) -> bool:
     )
 
 
+def _already_extracted(
+    element: pyparsing.ParserElement, lookup: ConverterState
+) -> bool:
+    return any(element.name == diag.kwargs["name"] for diag in lookup.diagrams.values())
```

Comment:
The `ConverterState` object indexes both of its dictionaries using the pyparsing element ID. So you can just check `return id(element) in lookup.diagrams` here.

Reply:
Ok, much simpler!